hermeto/core/package_managers/maven/main.py

```diff
@@ -0,0 +1,397 @@
+        )
+
+    def get_main_package(self) -> MavenComponentInfo:
+        """Get the main package as a MavenComponentInfo."""
+        main_info = self.main_package_info
+        purl = PackageURL(
+            type="maven",
+            namespace=main_info["groupId"],
+            name=main_info["artifactId"],
+            version=main_info["version"],
+        )
+
+        return MavenComponentInfo(
+            name=f"{main_info['groupId']}:{main_info['artifactId']}",
+            purl=str(purl),
+            version=main_info["version"],
+            scope="compile",
+            included=True,
+            missing_hash_in_file=None,
+        )
+
+    def get_sbom_components(self) -> list[MavenComponentInfo]:
+        """Get all dependencies as MavenComponentInfo objects."""
+        components = []
+        for dependency in self.dependencies:
+            if dependency.included:
+                components.append(dependency.to_component_info())
+        return components
+
+    def get_dependencies_to_download(self) -> dict[str, dict[str, Optional[str]]]:
+        """Get dictionary of dependencies to download."""
+        deps_to_download = {}
+        
+        for dependency in self.dependencies:
+            if dependency.included and dependency.resolved_url:
+                deps_to_download[dependency.resolved_url] = {
+                    "checksum": dependency.checksum,
+                    "checksum_algorithm": dependency.checksum_algorithm,
+                    "group_id": dependency.group_id,
+                    "artifact_id": dependency.artifact_id,
+                    "version": dependency.version,
+                }
+        
+        return deps_to_download
+
+
+def _get_maven_dependencies(
+    download_dir: RootedPath, deps_to_download: dict[str, dict[str, Optional[str]]]
+) -> dict[str, RootedPath]:
+    """Download Maven dependencies and return their local paths."""
+    download_paths = {}
+    files_to_download = {}
+    
+    for url, dep_info in deps_to_download.items():
+        # Create Maven directory structure: groupId/artifactId/version/
+        group_id = dep_info["group_id"] or ""
+        group_path = group_id.replace(".", "/")
+        artifact_path = f"{group_path}/{dep_info['artifact_id']}/{dep_info['version']}"
+        
+        # Extract filename from URL
+        parsed_url = urlparse(url)
+        filename = Path(parsed_url.path).name
+        
+        # Create local path
+        local_path = download_dir.join_within_root(artifact_path, filename)
+        local_path.path.parent.mkdir(parents=True, exist_ok=True)
+        
+        files_to_download[url] = local_path.path
+        download_paths[url] = local_path
+
+    # Download all files
+    if files_to_download:
+        asyncio.run(
+            async_download_files(
+                files_to_download,
+                get_config().concurrency_limit,
+            )
+        )
```

Comment:
I don't see any deduplication here. We just download everything based on `groupId/artifactId/version/` . What if you have:
```
Foo1 -> Baz
Foo2 -> Bar -> Baz
Foo3 -> FooBar -> Bar -> Baz
```
how many copies of `Baz` are we going to fetch? If this is anything like NPM's `node_modules` directory explosion then I think we'll need to deduplicate and only download a single copy and link it properly where needed to save bandwidth since it looks like we're in full control over this process, i.e. we're not relying on any native tooling underneath (e.g. `cargo`).